src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.diff.sql

```diff
@@ -305,3 +307,197 @@ AS
         
     COMMIT TRANSACTION
 GO
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     DeleteInstance
+--
+-- DESCRIPTION
+--     Removes the specified instance(s) and places them in the DeletedInstance table for later removal
+--
+-- PARAMETERS
+--     @cleanupAfter
+--         * The date time offset that the instance can be cleaned up.
+--     @createdStatus
+--         * Status value representing the created state.
+--     @studyInstanceUid
+--         * The study instance UID.
+--     @seriesInstanceUid
+--         * The series instance UID.
+--     @sopInstanceUid
+--         * The SOP instance UID.
+/***************************************************************************************/
+ALTER PROCEDURE dbo.DeleteInstance (
+    @cleanupAfter       DATETIMEOFFSET(0),
+    @createdStatus      TINYINT,
+    @studyInstanceUid   VARCHAR(64),
+    @seriesInstanceUid  VARCHAR(64) = null,
+    @sopInstanceUid     VARCHAR(64) = null
+)
+AS
+    SET NOCOUNT ON
+    SET XACT_ABORT ON
+
+    BEGIN TRANSACTION
+
+    DECLARE @deletedInstances AS TABLE
+        (StudyInstanceUid VARCHAR(64),
+            SeriesInstanceUid VARCHAR(64),
+            SopInstanceUid VARCHAR(64),
+            Status TINYINT,
+            Watermark BIGINT)
+
+    DECLARE @studyKey BIGINT
+    DECLARE @seriesKey BIGINT
+    DECLARE @instanceKey BIGINT
+    DECLARE @deletedDate DATETIME2 = SYSUTCDATETIME()
+
+    -- Get the study, series and instance PK
+    SELECT  @studyKey = StudyKey, @seriesKey =
+        CASE @seriesInstanceUid
+            WHEN null THEN null
+            ELSE SeriesKey
+        END,
+        @instanceKey = 
+        CASE @sopInstanceUid
+            WHEN null THEN null
+            ELSE InstanceKey
+        END
+    FROM    dbo.Instance
+    WHERE   StudyInstanceUid = @studyInstanceUid
+    AND     SeriesInstanceUid = ISNULL(@seriesInstanceUid, SeriesInstanceUid)
+    AND     SopInstanceUid = ISNULL(@sopInstanceUid, SopInstanceUid)
+
+    -- Delete the instance and insert the details into DeletedInstance and ChangeFeed
+    DELETE  dbo.Instance
+        OUTPUT deleted.StudyInstanceUid, deleted.SeriesInstanceUid, deleted.SopInstanceUid, deleted.Status, deleted.Watermark
+        INTO @deletedInstances
+    WHERE   StudyInstanceUid = @studyInstanceUid
+    AND     SeriesInstanceUid = ISNULL(@seriesInstanceUid, SeriesInstanceUid)
+    AND     SopInstanceUid = ISNULL(@sopInstanceUid, SopInstanceUid)
+
+    IF (@@ROWCOUNT = 0)
+    BEGIN
+        THROW 50404, 'Instance not found', 1;
+    END
+
+    -- Deleting indexed instance tags
+    DELETE
```

Comment:
when delete series, there could be multiple instances be deleted as well, I'm thinking should we remove instances stored in @deletedInstances instead.

This delete series level tag index data, but not instance level.